src/frontend/src/components/common/AppActivityManager/index.tsx

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  authLogoutHandler,
+  isAuthenticated,
+  refreshAuthTokens,
+  TokenStorageKeys,
+} from '../../../utils/auth';
+import { AppStorage } from '../../../utils/storage';
+import Modal from '../Modal';
+import { addEventListeners, removeEventListeners } from './events';
+
+const AppActivityManager = (): React.ReactElement => {
+  const navigate = useNavigate();
+
+  const [isWarningModalOpen, setWarningModalOpen] = useState(false);
+
+  useEffect(() => {
+    // These configs can come from OpenShift in the future
+    const config = {
+      refreshInterval: 1 * 60, // refresh token every one minute
+      inactivityTimeout: 14 * 60, // show warning modal at 14th minute
+      warningModalTimeout: 1 * 60, // if not acted on warning modal, auto logout after one minute
+    };
+
+    /**
+     * Refresh token Interval
+     * Refresh the authentication tokens every "refreshInterval" seconds [initial value - every one minute]
+     * Continue refreshing until user is logged out
+     *
+     * If any of the opened application tabs has refreshed the token in last one minute [or "refreshInterval" time ago ]
+     * then skip the refresh from the tab to avoid duplicate callings of refresh token
+     *
+     * This interval is stopped when the user logs out of the application
+     */
+    const interval = setInterval(() => {
+      const tokensLastRefreshedAt = AppStorage.getItem(
+        TokenStorageKeys.TOKENS_LAST_REFRESHED_AT,
+      );
+      const now = +new Date();
```

Comment:
Is `+new` a typo? I can't find any docs for it.

Reply:
ohh no, `+` is a valid javascript syntax and is short and equivalent to the `Number` method, which is used for converting types to numbers [type casting]. In this case we are converting a new date [current] type to a number, which is millis elapsed since epoch.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number

Reply:
Here's unary plus docs as well for ref

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus

Reply:
That's so cool! Thanks for the explanation and the doc refs 😃🧠